cmd/agent-sandbox-controller/main.go

```diff
@@ -56,13 +67,42 @@ func main() {
 
 	ctrl.SetLogger(zap.New(zap.UseFlagOptions(&opts)))
 
+	// Importing net/http/pprof registers handlers on the global DefaultServeMux.
+	// Reset it to avoid accidentally exposing pprof via any server that uses the default mux.
+	http.DefaultServeMux = http.NewServeMux()
+
 	scheme := controllers.Scheme
 	if extensions {
 		utilruntime.Must(extensionsv1alpha1.AddToScheme(scheme))
 	}
 
+	metricsOpts := metricsserver.Options{
+		BindAddress: metricsAddr,
+	}
+	if enablePprof || enablePprofDebug {
+		setupLog.Info("pprof enabled", "debug", enablePprofDebug)
+		metricsOpts.ExtraHandlers = map[string]http.Handler{
+			"/debug/pprof/profile": http.HandlerFunc(pprof.Profile),
+		}
+		if enablePprofDebug {
+			setupLog.Info("pprof debug endpoints enabled", "debug", enablePprofDebug)
+			runtime.SetBlockProfileRate(1)     // 1 = sample all events
```

Comment:
`runtime.SetBlockProfileRate(1)` configures the runtime to sample *every* blocking event (1ns resolution). This introduces massive overhead. Even for a debug flag, this is likely too aggressive and might render the controller unresponsive under load. Consider using a higher value (e.g., `1000000` for 1ms) or making it configurable.